site/docs/hosting/vps.md

```diff
@@ -2,9 +2,84 @@
 
 A virtual private server, mostly known as VPS, is a virtual machine running in the cloud with its users having the full control of its system.
 
-In this guide, you'll learn about various methods of running your bot in a VPS, keeping it online 24/7, making it run automatically when your VPS boots and restart on crashes.
+## Preparing Your Code
 
-## systemd
+To get started, you can use the bot shown in the [example of using a session](../plugins/session.md#example-usage).
+
+In the following, we will assume that the bot runs on Node.js
+Why only Node.js and not Deno or is it better for both runtimes?
+The answer is simple: the Deno pipeline would be very simple.
+A Node.js pipeline is complicated by the additional task of building and setting up communication between the two tasks.
+So a CI/CD pipeline for Node.js would be a better example.
+
+## Server Rental
+
+We chose [Hostinger](https://hostinger.com/) as our host.
+There are several reasons for this:
+
+1. It is multilingual, so you will find it convenient and easy to use.
+2. It has a very rich documentation that thoroughly covers all the problem areas that a beginner may encounter.
+3. It offers powerful servers at fairly low prices.
```

Comment:
How about telling users instead that they can pick whichever hosting provider they like? What they need to know is how to host on a VPS, and we should like to our [comparison](https://grammy.dev/hosting/comparison.html#vps). I don't see why we need to push them in one direction or the other.

Naturally, we can still guide them through the process for this one hosting provider in particular. So if they really don't know what to do, we can at least give them some guidance.

Also, what about proximity to Amsterdam?

Reply:
From the point of view of the article's user, I would have been happier if the article contained a guide from the actual server lease to CI/CD. This would be useful at least from an overview point of view of how the server lease stage works. If I need to deal with server lease separately, then why do I need this guide? I can deal with server leasing separately, and with systemd/pm2 and CI/CD separately. But we provide the minimum knowledge necessary for bots.

Instead, we could simply leave a note that the user can choose any provider, and the relevant part of the article serves only as an overview.

Reply:
What do you mean by "proximity to Amsterdam"?

If this refers to the location of the server, I don't think it plays any role for bots, because the user cannot know in advance which Telegram server will serve his bot.

The same goes for the API's own telegram servers. Although in this case, it is logical to place them, if not on the same server, then at least in the same geographical area.

Reply:
Most people use the Bot API server hosted by Telegram, which is in Amsterdam only.

Reply:
> Most people use the Bot API server hosted by Telegram, which is in Amsterdam only.

Huh, that's interesting. I didn't know that fact. Then it's really worth adding a recommendation.

Reply:
I still find this advertisement a little peculiar. It sounds like we are affiliated with them.

Reply:
What exactly do you not like? Should I completely remove the description of these benefits?

Reply:
IMO yes. If you want to compare it, try being less biased (list both pros and cons). But in the end, I already don't see why we need to advertise the service at all. We use it in this tutorial and that's it, the user has no choice but to read it.

We already say that they can use any service. We also have a comparison table that we link to.